resources/aws_ec2.go

```diff
@@ -85,6 +93,29 @@ const (
 	waitTimeout = 400
 )
 
+// snsCanonicalTemplate is a template that formats sns messages according to
+// amazon's canonical message format needed to check message signatures.
+// We use printf for the URL so we don't escape special characters.
+const snsCanonicalTemplate = `Message
```

Comment:
sadly I think this is a good use case for:

`var str string
str += fmt.Sprintf(XXX)
if foo {
   str += fmt.Sprintf(XXX)

}
```

Also please add a link to the docs about this so people don't think we're insane. (amazon is insane)